common-core/src/main/java/com/amplifyframework/notifications/pushnotifications/NotificationPayload.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.notifications.pushnotifications
+
+import android.content.Intent
+import android.os.Parcelable
+import kotlinx.parcelize.IgnoredOnParcel
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+open class NotificationContentProvider : Parcelable {
+    @Parcelize
+    data class FCM(val content: Map<String, String>) : NotificationContentProvider()
+}
+
+@Parcelize
+open class NotificationPayload(
+    val contentProvider: NotificationContentProvider?,
```

Comment:
Looking at this, I think it should be non-null

Reply:
I intentionally kept it non null. Most of our classes that has builders have internal constructor. But NotificationPayload constructor is need for PinpointNotificationPayload. So if customer tries to instantiate NotificationPayload without builder, at least content provider should be required.

Reply:
Do you mean you kept it nullable? It sounds like we are saying the same thing, that contentProvider should be a required non-null field. I'd rather have the builder fail and provide a null NotificationPayload than allow a NotificationPayload that is completely empty.

Reply:
Lets not go with a failing builder then, but one that contains contentProvider in the builder param instead of as a builder option. Its not a nullable field and we shouldn't treat it as such.